tc/src/main/java/com/epam/tc/web/controller/CoursesController.java

```diff
@@ -15,11 +29,165 @@
     private CourseService courseService;
     @Autowired
     private AuthenticatedUser authenticatedUser;
+    @Autowired
+    private UserService userService;
+
+    private static final String ID = "id";
 
     @RequestMapping(value = {"/courses", "/*"}, method = RequestMethod.GET)
     public Model courses(Model model) {
-        model.addAttribute("user", authenticatedUser.getUserName());
+        model.addAttribute("user", authenticatedUser.getUserEmail());
         model.addAttribute("courses", courseService.getAll());
+        model.addAttribute("person", userService.getUserByEmail(authenticatedUser.getUserEmail()));
         return model;
     }
+
+    private Course getCourse(String id) {
+        try {
+            int courseId = Integer.parseInt(id);
+            Course course = courseService.getById(courseId);
+            if (course == null) {
+                throw new CourseNotFoundException("Course with id: " + id + "not found");
+            }
+            return course;
+        } catch (NumberFormatException nfe) {
+            throw new IdParsingException("Cannot parse to int courseId: " + id, nfe);
+        }
+    }
+
+    @RequestMapping(value = {"/courses/{id}"}, method = RequestMethod.GET)
+    public ModelAndView details(@PathVariable(ID) String id) {
+        ModelAndView mav;
+        mav = new ModelAndView("courseDetails");
+        mav.addObject("course", getCourse(id));
+
+        return mav.addObject("user", authenticatedUser.getUserEmail());
+    }
+
+    @RequestMapping(value = "/courses/create", method = RequestMethod.GET)
+    public ModelAndView createCourse() {
+        return new ModelAndView("create").addObject("user", authenticatedUser.getUserEmail());
+    }
+
+    @RequestMapping(value = "/courses/create", method = RequestMethod.POST)
+    public void createCourse(final HttpServletRequest req,
+            final HttpServletResponse resp) throws IOException {
+
+        Course course = new Course(
+                req.getParameter("titleField"),
+                req.getParameter("descriptionField"),
+                req.getParameter("linksField"),
+                userService.getUserByEmail(authenticatedUser.getUserEmail()));
+        courseService.create(course);
+        resp.sendRedirect("/courses");
+    }
+
+    @RequestMapping(value = "/courses/{id}/update", method = RequestMethod.GET)
+    public ModelAndView printForUpdateCourse(@PathVariable(ID) String id) {
+        ModelAndView mav = new ModelAndView("troublePage");
+        Course course = getCourse(id);
+        String ownerEmail = course.getOwner().getEmail();
+        if (ownerEmail.equals(authenticatedUser.getUserEmail())) {
+            mav = new ModelAndView("update");
+            mav.addObject("course", course);
+        }
+        return mav.addObject("user", authenticatedUser.getUserEmail());
```

Comment:
you have to add user in most part of your methods. I think, this can be moved in a method annotated with @ModelAttribute